cinema_project/base_app/views.py

```diff
@@ -1,3 +1,156 @@
-from django.shortcuts import render
+import requests
+import json
 
-# Create your views here.
+from django.shortcuts import render, redirect, HttpResponse
+from django.contrib.auth import authenticate, login
+from django.contrib import messages
+from django.contrib.auth.forms import UserCreationForm
+from django.conf import settings
+from django.core.paginator import Paginator
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django.contrib.auth import get_user_model
+from django.template.loader import render_to_string
+from django.contrib.sites.shortcuts import get_current_site
+from ratelimit.decorators import ratelimit
+from django.utils.encoding import force_bytes, force_str
+
+from .models import ContactMessages, Schedule, Movie
+from .utils import next_days, today
+from .forms import SignupForm
+from .token import account_activation_token
+
+
+def login_user(request):
+    if request.method == 'POST':
+        email = request.POST['email']
+        password = request.POST['password']
+        user = authenticate(request, username=email, password=password)
+        if user is not None:
+            login(request, user)
+            return redirect('bookings')
+        else:
+            messages.success(request, message='There was an error logging in')
+            HttpResponse('Invalid credentials')
+            return redirect('login_page')
+    else:
+        return render(request, 'login_page.html')
+
+
+def homepage(request):
+    return render(request, 'homepage.html')
+
+
+def signup(request):
+    if request.method == "POST":
+        form = SignupForm(request.POST)
+        print('Method was POST')
+        if form.is_valid():
+            print('Form was valid')
+            user = form.save(commit=False)
+            user.is_active = False
+            user.save()
+
+            current_site = get_current_site(request)
+            message = render_to_string('acc_activate_email.html', {
+                'user': user,
+                'domain': current_site.domain,
+                'uid': urlsafe_base64_encode(force_bytes(user.pk)),
+                'token': account_activation_token.make_token(user),
+            })
+            activation_form_data = {
```

Comment:
Can this be done without hardcoding these values?

Reply:
Yes, by splitting the send_contact_email into a more general purpose function, will leave this as a future feature if it's not too important, what do you think ?